### ISSUE TITLE
'method' object not subscriptable when loading a config file with 'strategy' argument supplied to Trainer

### ISSUE THREAD

**Issue by TKassis** [Lightning-AI/lightning#15240]
### Bug description

I use Weights & Biases with PL all the time and it works great. 

```python
    wandb_logger = WandbLogger(
        project="Project Name",
        config="configs/config.yaml",
        log_model=True,
        mode="offline",
    )
    config = wandb_logger.experiment.config
    wandb_logger.experiment.log_code(".")
    model_name = wandb_logger.experiment.name

    # Load dataset
    dataset = MSDataModule(
        config["dataset_path"],
        config["batch_size"],
    )
```
The above works without any problems when the 'strategy' argument is not supplied to Trainer:

```python
    trainer = pl.Trainer(
        accelerator="gpu",
        devices=[1, 2, 3],
        precision=16,
        max_epochs=config["epochs"],
        callbacks=[checkpoint_callback, lr_monitor, model_summary],
        gradient_clip_val=1.0,
        logger=wandb_logger,
    )
```
But if I manually specificy the strategy (and it doesn't matter what it is) as follows:

```python
    trainer = pl.Trainer(
        accelerator="gpu",
        devices=[1, 2, 3],
        precision=16,
        strategy='ddp'
        max_epochs=config["epochs"],
        callbacks=[checkpoint_callback, lr_monitor, model_summary],
        gradient_clip_val=1.0,
        logger=wandb_logger,
    )
```
I get an error saying:

Exception has occurred: TypeError
**_'method' object is not subscriptable
  File "/home/......py", line 201, in <module>
    config["dataset_path"],_**

```

I thought for there was another variable in the namespace when I specify 'strategy' also called config, but even if I change the W&B config variable name to something else I get the same error.

### Environment

```

#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow): Trainer
#- PyTorch Lightning Version (e.g., 1.5.0): 1.7.7
#- PyTorch Version (e.g., 1.10): 1.12
#- Python version (e.g., 3.9): 3.9
#- OS (e.g., Linux): Linux
#- CUDA/cuDNN version: 11.6

```

**Comment by rohitgr7** [comment]
maybe you are missing a comma here:
```py
    trainer = pl.Trainer(
        accelerator="gpu",
        devices=[1, 2, 3],
        precision=16,
        strategy='ddp' # <- hereeeeeee
        max_epochs=config["epochs"],
        callbacks=[checkpoint_callback, lr_monitor, model_summary],
        gradient_clip_val=1.0,
        logger=wandb_logger,
    )
```

**Comment by TKassis** [comment]
Good find! But no that was just a typo here because I typed it in manually. My actual code has the comma :-)

**Comment by rohitgr7** [comment]
can you share the complete stack trace?

**Comment by awaelchli** [comment]
@TKassis The error message shows that this happens in `/home/timothy/Documents/CelestialAI/celestialai/train_foundation.py`. This is your code, right? Please take a look at line 201 in that file. The variable `config` seems to be a method, but the author of the code is accessing it as a dictionary.

**Comment by TKassis** [comment]
config is actually a dictionary, but for some reason when I used the 'strategy' argument suddenly it becomes interpreted as a method. I couldn't fix the issue, I think it might be WandB related. I ended up just not using the returned dictionary from WandB. It might be something related with using WandB with multiple GPUs (although no issue with the default 'ddp' strategy unless explicitly mentioned). Anyway, I'm closing this for now as I found a workaround.
